lib/SPIRV/SPIRVToOCL20.cpp

```diff
@@ -135,11 +135,29 @@ void SPIRVToOCL20::visitCallSPIRVControlBarrier(CallInst *CI) {
           return cast<ConstantInt>(Args[I])->getZExtValue();
         };
         auto ExecScope = static_cast<Scope>(GetArg(0));
-        auto MScope = static_cast<Scope>(GetArg(1));
-        auto Sema = mapSPIRVMemSemanticToOCL(GetArg(2));
+        auto MemScope = static_cast<Scope>(GetArg(1));
+
         Args.resize(2);
-        Args[0] = getInt32(M, Sema.first);
-        Args[1] = getInt32(M, rmap<OCLScopeKind>(MScope));
+        if (auto Arg = dyn_cast<ConstantInt>(Args[2])) {
+           auto Sema = mapSPIRVMemSemanticToOCL(Arg->getZExtValue());
```

Comment:
Indentation seems off by one here (see also the Travis build).
It would be nice to also have a brief comment that says which cases the `if` is distinguishing.

Reply:
Fixed